packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/dataservice/DatasetResource.java

```diff
@@ -168,9 +178,30 @@ public Response getCsv(
 		@PathParam("id") final String id,
 		@DefaultValue("true") @QueryParam("wide_format") final Boolean wideFormat,
 		@DefaultValue("false") @QueryParam("data_annotation_flag") final Boolean dataAnnotationFlag,
-		@DefaultValue("50") @QueryParam("row_limit") final Integer rowLimit
+		@DefaultValue("50") @QueryParam("row_limit") final Integer rowLimit,
+		@DefaultValue("0") @QueryParam("binCount") final Integer binCount
 	) {
-		return proxy.getCsv(id, wideFormat, dataAnnotationFlag, rowLimit);
+		
+		log.info("Getting csv content");
+		Response returnResponse = proxy.getCsv(id, wideFormat, dataAnnotationFlag, rowLimit);
+		String rawCsvString = proxy.getCsv(id, wideFormat, dataAnnotationFlag, rowLimit).readEntity(String.class);
```

Comment:
You may want to have error handling here if the proxied service is down/returns a bad value or empty string, etc.  As an example, see `HomeResource.java` starting around line 84 and 105